File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.py
class Solution:
    def maxCount(self, banned: List[int], n: int, maxSum: int) -> int:
        Booleanlist=[False]*n
        for x in banned:
            if x <= n:
                Booleanlist[x-1]=True
        count=0 
        for i in range(0,n):
            if(not Booleanlist[i] and i+1 <=maxSum):
                count+=1
                maxSum-=(i+1)
            elif i+1 > maxSum:
                return count
        
        return count

File: 2554-maximum-number-of-integers-to-choose-from-a-range-i/2554-maximum-number-of-integers-to-choose-from-a-range-i.java
class Solution {
    public int maxCount(int[] banned, int n, int maxSum) {
        Boolean A[]=new Boolean [n];
        for(int i=0;i<n;i++)A[i]=false;
        for(int i=0;i<banned.length;i++){
            if(banned[i]<=n)A[banned[i]-1]=true;
        }
        int sum=0;
        for(int i=0;i<n;i++){
            if(!A[i] && (i+1)<=maxSum){
                maxSum-=(i+1);
                sum++;
            }else if((i+1)>maxSum)return sum;
        }
        return sum;
    }
}